src/preferences/preferences.cpp

```diff
@@ -1550,6 +1550,37 @@ const std::vector<game_config::server_info>& prefs::builtin_servers_list()
 	return pref_servers;
 }
 
+void prefs::add_game_preset(config& preset)
+{
+	if(preset.has_child(prefs_list::game_preset) && preset.all_children_count() == 1 && preset.attribute_count() == 0) {
+		int max = 0;
+		for(const auto& c : preferences_.child_range(prefs_list::game_preset)) {
+			max = std::min(max, c["id"].to_int());
+		}
+		preset.mandatory_child(prefs_list::game_preset)["id"] = max-1;
```

Comment:
Unless I'm mistaken, this key will be `-1` in all cases since `max` will always be 0. There's never any `id` key provided, and the `to_int` default is 0.

Reply:
Line 1560 is where it's writing the id key.

Reply:
Right, but it's being written to `min - 1`, where the value of `min` is determined by the as-yet non-existent `id` key.